store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/ApplyViewToElementsFunction.java

```diff
@@ -77,13 +84,31 @@ public ApplyViewToElementsFunction(final Map<String, Object> context) throws Gaf
             }
             // Validate the supplied context before using
             validate(context);
+
+            updateViewWithValidationFromSchema(context);
+
             this.context = Collections.unmodifiableMap(context);
         } catch (final Exception e) {
             throw new GafferCheckedException("Unable to create TemporaryResultsGraph", e);
         }
 
     }
 
+    private static void updateViewWithValidationFromSchema(final Map<String, Object> context) {
+        //Only do this for MapStore, not required for other stores.
+        if (MapStore.class.getName().equals(((Graph) context.get(TEMP_RESULTS_GRAPH)).getStoreProperties().getStoreClass())) {
```

Comment:
This seems like a lot of manual tweaking of the View. I think adding validation to the MapStore would be a cleaner solution. Is there a reason to this over that?

Reply:
Yes, Mapstore should be fixed and this code should be ripped out.

Reply:
In that case, could it be written in such a way that it is obviously temporary? Maybe within one block that can just be deleted when the time comes?

Reply:
It's all One method `updateViewWithValidationFromSchema()` the with sub methods that exist is for general Java readability and such. They will become highlighted and unused when you delete `updateViewWithValidationFromSchema`